android/app/src/main/java/com/androidstartupperfapp/SecondFragment.java

```diff
@@ -41,4 +43,12 @@ private void onSecondButtonClick(View view) {
       .findNavController(SecondFragment.this)
       .navigate(R.id.action_SecondFragment_to_FirstFragment);
   }
+
+  private void onStartReactActivityButtonClick(View view) {
+    // In a fragment, we can't use the current activity (MainActivity) class directly.
+    // However, we can use getActivity() to get the job done.
+    // https://stackoverflow.com/a/15478468/11455106
+    Intent intent = new Intent(getActivity(), ReactActivity.class);
+    startActivity(intent);
+  }
```

Comment:
Hmmm... I might have stumbled upon something good here:

- https://stackoverflow.com/questions/9646308/android-activity-starting-parallel
- https://stackoverflow.com/questions/12257810/android-activity-startactivity-on-two-parallel-activities

```suggestion
  private void onStartReactActivityButtonClick(View view) {
    // In a fragment, we can't use the current activity (MainActivity) class directly.
    // However, we can use getActivity() to get the job done.
    // https://stackoverflow.com/a/15478468/11455106
    Intent intent = new Intent(getActivity(), ReactActivity.class);
    intent.addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT);
    startActivity(intent);
  }
```

Alternatively, still go with only one activity and use fragments instead, as is discussed in the counterpoints here:

- https://stackoverflow.com/questions/23969155/application-with-2-activities-running-parallel#comment36929555_23969247

Reply:
I think I got this one wrong.

- https://stackoverflow.com/questions/10008879/intent-to-start-activity-but-dont-bring-to-front
- https://stackoverflow.com/questions/15292779/create-new-intent-in-background

Reply:
Wait i think you are onto something here that might solve the preload issue. Is there maybe a different way of navigating to different fragments?

Reply:
https://developer.android.com/guide/navigation/navigation-getting-started#add-navhost

This is probably what we are looking at. Theoretically with 1 Activity + multiple fragment setup, we will have eliminated the need to ensure the right activity is selected for ReactInstanceManager since there are always only one Activity

Reply:
Good to go through the whole Fragment section here as well
https://developer.android.com/guide/fragments